src/DynamoCoreWpf/Utilities/PreferencesPanelUtilities.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Linq;
+using System.Windows;
+using System.Windows.Controls;
+using Dynamo.Controls;
+using Dynamo.Utilities;
+using Dynamo.Wpf.Views;
+
+namespace Dynamo.Wpf.Utilities
+{
+    /// <summary>
+    /// Utilities for the Preferences panel
+    /// </summary>
+    static class PreferencesPanelUtilities
+    {
+        /// <summary>
+        /// This static method will open the Preferences panel in the specified tab (if is provided) and expand the specified expander (if is provided)
+        /// </summary>
+        /// <param name="mainWindow">This window needs to be the DynamoView</param>
+        /// <param name="location">Location in which the Preference panel will be shown</param>
+        /// <param name="tabName">Tab name in which the preference panel will be opened</param>
+        /// <param name="expanderName">Expander name that will be expanded (it has to be inside the tab)</param>
+        public static void OpenPreferencesPanel(Window mainWindow, WindowStartupLocation location, string tabName = "", string expanderName = "")
+        {
+            var preferencesWindow = new PreferencesView(mainWindow as DynamoView);
```

Comment:
We may run into a bug if there is already a preferences window opened or not? To be tested

Reply:
@QilongTang the Preferences panel is a modal window then I think is not possible to have two windows opened at the same time, unless you know a way to create to modal windows,

Reply:
@RobertGlobant20 Understood, user may not be able to interact with the UI to trigger this code you mean. Let's just keep this in mind since I see same problem related to PM term of use. This window need to be singleton. I dont think we need to change this PR for it though